File: solutions/python/math-geometry/complex-number-multiplication.py
# LeetCode 537 — Complex Number Multiplication (Medium)
# Category: Math & Geometry · Approach: Parse and multiply
# Time: O(1) | Space: O(1)
# Source: https://leetcode.com/problems/complex-number-multiplication/

def complex_number_multiply(num1: str, num2: str) -> str:
    def parse(s: str) -> tuple[int, int]:
        a, b = s.split("+")
        return int(a), int(b[:-1])

    a, b = parse(num1)
    c, d = parse(num2)
    real = a * c - b * d
    imag = a * d + b * c
    return f"{real}+{imag}i"

File: solutions/typescript/math-geometry/complex-number-multiplication.ts
// LeetCode 537 — Complex Number Multiplication (Medium)
// Category: Math & Geometry · Approach: Parse and multiply
// Time: O(1) | Space: O(1)
// Source: https://leetcode.com/problems/complex-number-multiplication/

function complexNumberMultiply(num1: string, num2: string): string {
  const parse = (s: string): [number, number] => {
    const [a, b] = s.split("+");
    return [parseInt(a, 10), parseInt(b, 10)];
  };
  const [a, b] = parse(num1);
  const [c, d] = parse(num2);
  const real = a * c - b * d;
  const imag = a * d + b * c;
  return `${real}+${imag}i`;
}
